### PR TITLE
Fix Axis/Legend Schema + Make them respect Config 

### LINKED ISSUE #2683: Default axis/legend rules may not respect config
...since they outputs hardcode values and do not read config when determining rules.

- [x] axis
- [x] legend

### PR DESCRIPTION
**Pull request by kanitw**
Fix #2683 

- `config.axis*.grid`  now works

image

- `config.axis*.labelAngle`  now works

image

### CODE REVIEW

**Comment by arvind** [comment]
Regenerate examples?

**Comment by kanitw** [comment]
Done

File: src/compile/axis/encode.ts [hunk 1]
```diff
@@ -67,13 +70,16 @@ export function labelAngle(axis: Axis, channel: Channel, fieldDef: FieldDef<stri
   }
   return undefined;
 }
-export function labelAlign(angle: number, orient: AxisOrient) {
-  if (angle && angle > 0) {
-    if (angle > 180) {
+
+export function labelAlign(angle: number, orient: AxisOrient): HorizontalAlign {
+  if (angle > 0) {
+    if (angle % 360 > 180) {
```
Comment (domoritz) [inline src/compile/axis/encode.ts]: Why not normalize the angle once?

Comment (kanitw) [reply src/compile/axis/encode.ts]: I think it's not relevant to this PR and it's not easy to see this large diff here so I'll just merge.

**Pull request merged**